src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java

```diff
@@ -31,29 +34,34 @@ public KorpergrosseComposition toComposition(Observation observation) {
             return null;
         }
 
-        // references:
-        // https://ckm.highmed.org/ckm/templates/1246.169.1038
-        // https://tools.openehr.org/designer/?code=39ddef9b1e993dc441ff#/designer/repos/num-2021-01-11
-        // https://simplifier.net/ForschungsnetzCovid-19/BodyHeight/~overview
-        // https://simplifier.net/forschungsnetzcovid-19/observation-example-body-height
-
         KorpergrosseComposition result = new KorpergrosseComposition();
         GrosseLangeObservation grosseLangeObservation = new GrosseLangeObservation();
 
-        DateTimeType fhirEffectiveDateTime;
+
+        ZonedDateTime fhirEffectiveDateTime = null;
         try {
-            // default for every observation
-            fhirEffectiveDateTime = observation.getEffectiveDateTimeType();
 
-            //BSa Wie erscheint das im Mapping? Und wo finde ich das Referenzmodell mit diesen Infos zu openEHR?
-            grosseLangeObservation.setTimeValue(fhirEffectiveDateTime.getValueAsCalendar().toZonedDateTime());
-            grosseLangeObservation.setOriginValue(fhirEffectiveDateTime.getValueAsCalendar().toZonedDateTime()); // mandatory
+            if(observation.hasEffectiveDateTimeType()) {
```

Comment:
Move into seperate methdo pls

Reply:
I made many more methods and I wonder how you like it. The try-catch from the second comment you gave me is removed now. Is that ok?